releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh

```diff
@@ -0,0 +1,176 @@
+EXPORT_CONFIG_POST_ENDPOINT="snowowl/snomed-ct/v3/exports"
+
+# The address where the branch get endpoint can be found.
+BRANCH_GET_ENDPOINT="snowowl/snomed-ct/v3/branches"
+
+# The input data for the export config.
+EXPORT_CONFIG_POST_INPUT=""
+
+# The uuid of the export request given by Snow Owl.
+EXPORT_UUID=""
+
+# The input data for the code system version create config.
+CODE_SYSTEM_VERSION_CREATE_POST_INPUT=""
+
+# The initial name of the export.
+EXPORT_FILE_NAME="snow_owl_delta_export"
+
+# The renamed version of the export file.
+RENAMED_EXPORT_FILE="snow_owl_delta_export"
+
+# Calculates the current location of this specific bash script (works until the path is not a symlink)
+SCRIPT_LOCATION="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+usage() {
+
+	cat <<EOF
+NAME:
+	Automated delta export script. The export file will be saved where the script is located. By defining a target envrionemnt this script doesn't need to be present on the server.
+OPTIONS:
+	-h
+		Show this help
+	-u
+		Define a username with privileges to the Snow Owl REST API
+	-p
+		Define the password for the above user
+    -t
+        Target environment which server the export should be initiated on
+NOTES:
+	This script can be used to initiate a delta export job that will run an export from a snow owl server every day at 20:00 server time and it will save it to the folder the script is within.
+	Mandatory variables:
+		- SNOW OWL user to use for the snapshot export
+	    - SNOW OWL password for the above user
+        - The target enviroment where the Snow Owl server runs
+EOF
+
+}
+
+validate_variables() {
+    check_if_empty "${SNOW_OWL_USER}" "Snow Owl username must be specified"
+	check_if_empty "${SNOW_OWL_PASSWORD}" "Snow Owl password must be specified"
+	check_if_empty "${TARGET_FOLDER}" "Target folder must be specified"
+    check_if_empty "${TARGET_ENVIRONMENT}" "Target environment must be specified"
+
+    if [[ ! -d "${TARGET_FOLDER}" ]]; then
+        echo "Creating target folder"
+        mkdir "${TARGET_FOLDER}"
+    fi
+
+}
+
+check_if_empty() {
+	if [[ -z "$1" ]]; then
+		echo "$2"
+		exit 1
+	fi
+}
+
+export_delta() {
+    echo "Creating snapshot export config"
+    EXPORT_CONFIG_POST_INPUT='{"branchPath":"MAIN","codeSystemShortName":"SNOMEDCT","type":"DELTA"}'
+    EXPORT_UUID="$(curl -u "${SNOW_OWL_USER}:${SNOW_OWL_PASSWORD}" -i -X POST -H "Content-type: ${MEDIA_TYPE}" "http://${TARGET_ENVIRONMENT}:8080/${EXPORT_CONFIG_POST_ENDPOINT}" -d "${EXPORT_CONFIG_POST_INPUT}" | tr -d '\r' | sed -En 's/^Location: (.*)/\1/p')"
+    download_delta
+}
+
+download_delta() {
+    echo "EXPORT_UUID: ${EXPORT_UUID}"
+    EXPORT_DOWNLOAD_GET_ENDPOINT="${EXPORT_UUID}/archive"
+
+    echo "Downloading snapshot export with UUID: ${EXPORT_DOWNLOAD_GET_ENDPOINT}"
+
+    curl -u "${SNOW_OWL_USER}:${SNOW_OWL_PASSWORD}" -X GET -H "Accept: ${ACCEPT_HEADER}" -ko "${EXPORT_FILE_NAME}" "${EXPORT_DOWNLOAD_GET_ENDPOINT}" 
```

Comment:
You can immediately download the file with a timestamped name with `"${EXPORT_FILE_NAME}_${DATE}"`